src/engine/GroupBy.cpp

```diff
@@ -872,9 +866,9 @@ GroupBy::extractValuesDirectlyFromMap(
       getExecutionContext()->getAllocator());
 
   // Store aggregate results in table and in vector.
-  for (size_t j = evaluationContext._beginIndex;
-       j < evaluationContext._endIndex; j++) {
-    auto& val = (*resultTable)(j, 0);
+  decltype(auto) groupValues = resultTable->getColumn(0);
```

Comment:
In a future PR you can use the same thing for the `outCol` of the result table...